### PR TITLE
Season Class

### PR DESCRIPTION
**Pull request by ErickHeyl**
Added a simple season class to be extended once TvShow / Episodes is
done

### CODE REVIEW

File: PlotTalk/Src/season.cpp [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include "season.h"
+
+/**
+ * @brief Season:: a TvShow will have a list of Seasons, and each season will have a list of episodes
+ * @param number of season
+ * @param a list of episdoes for that season
+ */
+Season::Season(QString seasonNumber,QList episdoes)
```
Comment (haleyminshell) [inline PlotTalk/Src/season.cpp]: episodes is misspelled as "episdoes". Not a big deal but just noticed.

File: PlotTalk/Src/season.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#define SEASON_H
+#include <QString>
+#include <Qlist>
+
+
+class Season
+{
+public:
+    QString seasonNumber;
+    Qlist<Episodes> episdoes;
```
Comment (haleyminshell) [inline PlotTalk/Src/season.h]: "episdoes" -> episodes

File: PlotTalk/Src/season.cpp [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    this->episdoes = episdoes;
+}
+/**
+ * @brief Season:: a TvShow will have a list of Seasons, and each season will have a list of episodes
+ * @param number of season
+ * @return int number of episodes in episdoes list
+ */
+int Season::numberOfEpisodes()
+{
+    return this->episdoes.size();
```
Comment (allanhaywood) [inline PlotTalk/Src/season.cpp]: Optional as this is more of a style thing:

this-> is only needed when the names are hidden by local method parameters.
You should be able to use:
return episodes.size();

**Review by allanhaywood — CHANGES_REQUESTED** [review]
Please add unit tests for this class.

File: PlotTalk/PlotTalk.pro [hunk 4]
```diff
@@ -28,7 +28,9 @@ HEADERS += \
     Src/reaction.h \
     Src/reply.h \
     Src/review.h \
-    Src/comment.h
+    Src/comment.h \
+    Src/season.h \
+    Test/testseason.h
```
Comment (allanhaywood) [inline PlotTalk/PlotTalk.pro]: This needs to be moved into the test section.

Comment (allanhaywood) [reply PlotTalk/PlotTalk.pro]: See: https://docs.google.com/document/d/1DiG4xUtyUR_7fSkczaVVn59AJ-xZpCTzNsWXZCk96vQ/edit#heading=h.xqfezzdmgo85
for details.

**Review by allanhaywood — CHANGES_REQUESTED** [review]
Thank you for adding the tests. There are a couple more steps to do to get them to run.

Move the references to the tests in the pro file to the test section, and add references to the test in testmain.cpp.

Details can be found in this doc: https://docs.google.com/document/d/1DiG4xUtyUR_7fSkczaVVn59AJ-xZpCTzNsWXZCk96vQ/edit#heading=h.xqfezzdmgo85

**Review by allanhaywood — APPROVED** [review]

**Pull request merged**